### PR TITLE
SG for Chips DBs application access

### PR DESCRIPTION
**Pull request by willwatters**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,15 +4,14 @@ repos:
     hooks:
       - id: check-merge-conflict
   - repo: git://github.com/antonbabenko/pre-commit-terraform
-    rev: v1.50.0
```
Comment (MartyFox) [inline .pre-commit-config.yaml:7]: Need to maintain the newer version

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -4,15 +4,14 @@ repos:
...
+    rev: v1.43.0
     hooks:
       - id: terraform_fmt
       - id: terraform_docs
-        args: ["--args=--sort-by required"]
```
Comment (MartyFox) [inline .pre-commit-config.yaml:11]: Shouldnt remove this.

File: groups/chips-db/variables.tf [hunk 3]
```diff
@@ -264,3 +264,13 @@ variable "netapp_ips" {
...
   description = "List of Netapp IP addresses to use for iscsi discovery."
 }
+
+
+# ------------------------------------------------------------------------------
+# CHIPS DB SG variables
+# ------------------------------------------------------------------------------
+variable "chips_db_sg" {
+  type        = list(any)
+  description = "List of CHIPS DB SGs"
```
Comment (MartyFox) [inline groups/chips-db/variables.tf:274]: SGs should be security groups in the description, make it easier for others to understand.

File: groups/chips-rep-db/ec2.tf [hunk 4]
```diff
@@ -40,20 +33,13 @@ module "db_ec2_security_group" {
...
-      source_security_group_id = data.aws_security_group.chips_oltp.id
-    },
+  ingress_with_source_security_group_id = [for group in local.source_security_group_id :
     {
       from_port                = 1521
       to_port                  = 1522
       protocol                 = "tcp"
-      description              = "Oracle DB Staffware Security Group"
-      source_security_group_id = data.aws_security_group.staffware.id
+      description              = "Oracle DB CHIPS Security Group"
```
Comment (MartyFox) [inline groups/chips-rep-db/ec2.tf:41]: CHIPS -> CHIPS REP

File: groups/chips-rep-db/variables.tf [hunk 5]
```diff
@@ -264,3 +264,12 @@ variable "netapp_ips" {
   type        = list(string)
   description = "List of Netapp IP addresses to use for iscsi discovery."
 }
+
+# ------------------------------------------------------------------------------
+# CHIPS DB SG variables
+# ------------------------------------------------------------------------------
+variable "chips_db_sg" {
+  type        = list(any)
+  description = "List of CHIPS DB SGs"
```
Comment (MartyFox) [inline groups/chips-rep-db/variables.tf:273]: Same as above

File: groups/staffware-db/ec2.tf [hunk 6]
```diff
@@ -15,41 +15,14 @@ module "db_ec2_security_group" {
...
-    },
+
+  ingress_with_source_security_group_id = [for group in local.source_security_group_id :
     {
       from_port                = 1521
       to_port                  = 1522
       protocol                 = "tcp"
-      description              = "Oracle DB CHIPS Rep Security Group"
-      source_security_group_id = data.aws_security_group.chips_rep.id
+      description              = "Oracle DB CHIPS Security Group"
```
Comment (MartyFox) [inline groups/staffware-db/ec2.tf:24]: CHIPS -> Staffware

File: groups/staffware-db/profiles/heritage-live-eu-west-2/vars [hunk 7]
```diff
@@ -74,3 +74,12 @@ failover_approvers = [
 ]
 
 netapp_ips = ["10.84.9.97"]
+
+# CHIPS DB SGs
+chips_db_sg = [
```
Comment (MartyFox) [inline groups/staffware-db/profiles/heritage-live-eu-west-2/vars:79]: chips_db_sg -> staffware_db_sg, same for the comment

File: groups/staffware-db/profiles/heritage-staging-eu-west-2/vars [hunk 8]
```diff
@@ -74,3 +74,12 @@ failover_approvers = [
 ]
 
 netapp_ips = ["10.94.9.185"]
+
+# CHIPS DB SGs
```
Comment (MartyFox) [inline groups/staffware-db/profiles/heritage-staging-eu-west-2/vars:78]: Same as above

File: groups/staffware-db/variables.tf [hunk 9]
```diff
@@ -234,3 +234,12 @@ variable "netapp_ips" {
   type        = list(string)
   description = "List of Netapp IP addresses to use for iscsi discovery."
 }
+
+# ------------------------------------------------------------------------------
+# CHIPS DB SG variables
+# ------------------------------------------------------------------------------
+variable "chips_db_sg" {
```
Comment (MartyFox) [inline groups/staffware-db/variables.tf:241]: same as above for chips -> staffware naming

File: groups/chips-db/variables.tf (hunk 3 above)
Comment (willwatters) [reply groups/chips-db/variables.tf:274]: Changed to Security Groups

File: groups/chips-rep-db/ec2.tf (hunk 4 above)
Comment (willwatters) [reply groups/chips-rep-db/ec2.tf:41]: Changed to description              = "Oracle DB CHIPS REP Security Group"

File: groups/chips-rep-db/variables.tf (hunk 5 above)
Comment (willwatters) [reply groups/chips-rep-db/variables.tf:273]: Changed to description = "List of CHIPS DB REP Security Groups"

File: groups/staffware-db/ec2.tf (hunk 6 above)
Comment (willwatters) [reply groups/staffware-db/ec2.tf:24]: Changed to description              = "Oracle DB Staffware Security Group"

File: groups/staffware-db/variables.tf (hunk 9 above)
Comment (willwatters) [reply groups/staffware-db/variables.tf:241]: changed to description = "List of Staffware DB Security Groups"

File: groups/staffware-db/profiles/heritage-live-eu-west-2/vars (hunk 7 above)
Comment (willwatters) [reply groups/staffware-db/profiles/heritage-live-eu-west-2/vars:79]: Changed to staffware_db_sg

File: .pre-commit-config.yaml (hunk 1 above)
Comment (willwatters) [reply .pre-commit-config.yaml:7]: Having issues with this version so looked at other repos and they are using the following which works:

repos:
  - repo: git://github.com/pre-commit/pre-commit-hooks
    rev: v3.2.0
    hooks:
      - id: check-merge-conflict
  - repo: git://github.com/antonbabenko/pre-commit-terraform
    rev: v1.43.0
    hooks:
      - id: terraform_fmt
      - id: terraform_docs
      - id: terraform_tfsec
      - id: terraform_tflint
  - repo: git://github.com/dnephin/pre-commit-golang
    rev: v0.3.5
    hooks:
      - id: go-fmt
      - id: go-lint

File: .pre-commit-config.yaml (hunk 2 above)
Comment (willwatters) [reply .pre-commit-config.yaml:11]: Having issues with this version so looked at other repos and they are using the following which works:

repos:
  - repo: git://github.com/pre-commit/pre-commit-hooks
    rev: v3.2.0
    hooks:
      - id: check-merge-conflict
  - repo: git://github.com/antonbabenko/pre-commit-terraform
    rev: v1.43.0
    hooks:
      - id: terraform_fmt
      - id: terraform_docs
      - id: terraform_tfsec
      - id: terraform_tflint
  - repo: git://github.com/dnephin/pre-commit-golang
    rev: v0.3.5
    hooks:
      - id: go-fmt
      - id: go-lint

Comment (willwatters) [reply .pre-commit-config.yaml]: Added back in:

 - id: terraform_docs
        args: ["--args=--sort-by required"]

**Review by MartyFox — APPROVED** [review]

**Pull request merged**